src/gui/src/heatMap.cpp

```diff
@@ -583,43 +622,92 @@ void HeatMapRenderer::drawObjects(Painter& painter)
 
   const odb::Rect& bounds = painter.getBounds();
 
-  for (const auto& [bbox, map_pt] : datasource_.getMap()) {
-    if (!map_pt->has_value) {  // value not set so nothing to draw
+  // minimum box size is 2 pixels
+  const double min_dbu = 2.0 / painter.getPixelsPerDBU();
+
+  const double dbu_per_micron = datasource_.getBlock()->getDbUnitsPerMicron();
+  const int x_scale
+      = std::ceil(min_dbu / (datasource_.getGridXSize() * dbu_per_micron));
```

Comment:
microns, and it could, but when we save the display settings it's easier to do that in microns since when we open on a new design/technology the dbus might be different.

I think rewriting that would be a different PR as it's a bigger change.

Reply:
I think it would be good but am fine to separate it.